packages/contracts-periphery/contracts/universal/op-nft/OptimistInviter.sol

```diff
@@ -0,0 +1,274 @@
+     *                        signature is that the issuer may be using a ERC-1271 compatible
+     *                        contract wallet, where the recovered address is not the same as the
+     *                        issuer, or the signature is not an ECDSA signature at all.
+     * @custom:field nonce    Pseudorandom nonce to prevent replay attacks.
+     */
+    struct ClaimableInvite {
+        address issuer;
+        bytes32 nonce;
+    }
+
+    /**
+     * @notice Maps from hashes to whether or not they have been committed.
+     */
+    mapping(bytes32 => bool) public commitments;
+
+    /**
+     * @notice Maps from addresses to nonces to whether or not they have been used.
+     */
+    mapping(address => mapping(bytes32 => bool)) public usedNonces;
+
+    /**
+     * @custom:semver 1.0.0
+     *
+     * @param _inviteGranter      Address of the invite granter.
+     * @param _attestationStation Address of the AttestationStation contract.
+     */
+    constructor(address _inviteGranter, AttestationStation _attestationStation) Semver(1, 0, 0) {
+        INVITE_GRANTER = _inviteGranter;
+        ATTESTATION_STATION = _attestationStation;
+    }
+
+    /**
+     * @notice Initializes this contract, setting the EIP712 context.
+     *
+     *         When upgrading this contract, make sure to re-initialize and to update the version
+     *         in the EIP712 context to the up-to-date semver by calling __EIP712_init again. After
+     *         the EIP712 version is changed, any signatures that were issued off-chain but not
+     *         claimed yet, will no longer be accepted by the claimInvite function. Please make
+     *         sure to notify the issuers that they must re-issue their invite signatures.
+     *
+     * @param _name Contract name
+     */
+    function initialize(string memory _name) public initializer {
+        __EIP712_init(_name, version());
+    }
+
+    /**
+     * @notice Allows invite granter to set the number of invites an address has.
+     *
+     * @param _accounts    An array of accounts to update the invite counts of.
+     * @param _inviteCount Number of invites to set to.
+     */
+    function setInviteCounts(address[] calldata _accounts, uint256 _inviteCount) public {
+        // Only invite granter can grant invites
+        require(
+            msg.sender == INVITE_GRANTER,
+            "OptimistInviter: only invite granter can grant invites"
+        );
+
+        uint256 length = _accounts.length;
+
+        AttestationStation.AttestationData[]
+            memory attestations = new AttestationStation.AttestationData[](length);
+
+        for (uint256 i; i < length; ) {
+            // The granted invites are stored as an attestation from this contract on the
+            // AttestationStation contract. Number of invites is stored as a encoded uint256 in the
+            // data field of the attestation.
+            attestations[i] = AttestationStation.AttestationData({
```

Comment:
Unsure how I feel about using the ATST to store invite counts. No real benefit over storing internally. If anything, could be worth giving a single attestation representing the fact that the user *did* have invites but not bothering to use it to store state.

Reply:
Updated the contract to store a separate mapping of invite counts. The contract still issues a single attestation to show that the user was allowed to invite others, but the contract no longer depends on the data field to handle the invite count.

Reply:
Nice catch!